### PR TITLE
Issue #11735: When ACME fails to create a certificiate or keystore on…

### LINKED ISSUE #11733: acmeCA-2.0: Test with an empty keystore
Verify that we still get a certificate if the user configures an empty keystore.

For #9017 

FYI @jvanhill

### LINKED ISSUE #11735: acmeCA-2.0: Failures to generate keystore on startup due to configuration should be dynamically fixable
When generating the certificate and / or keystore on startup fails because of a configuration issue, we need to be able to fix the configuration to correct the issue. The feature should then generate a new certificate and insert it into the keystore.

### PR DESCRIPTION
**Pull request by jvanhill**
… initial startup due to configuration, ACME should generate the certificate upon updating the configuration to a valid state.

fixes #11735, fixes #11733

### CODE REVIEW

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
@acdemyers - Mind reviewing the SSL specific code? I updated the certificate creator interface to accept the keystore provider and type so it no longer needs to look at the file suffix.

**Comment by jvanhill** [comment]
> Updated the KeystoreService to allow me to look up a keystore configurations type and provider. I use this in ACME to create the keystore.

Actually, looks like I didn't end up using these methods. I can probably remove them.

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops [hunk 1]
```diff
@@ -154,9 +154,9 @@ CWPKI2032E=CWPKI2032E: The certificate subject alternative names in the default
 CWPKI2032E.explanation=The certificate is an invalid DER-encoded certificate or contains unsupported DER features.
 CWPKI2032E.useraction=Review the error message for details on the failure. Revoke or remove the invalid certificate.
 
-CWPKI2033E=CWPKI2033E: The ACME service could not find the certificate for the {0} alias in the {1} keystore. The error is ''{2}''.
-CWPKI2033E.explanation=A certificate authority signed certificate is stored in the default keystore and replaces any existing default certificate, but keystore or default certificate could not be accessed.
-CWPKI2033E.useraction=Review the error message for details on the failure.
+#CWPKI2033E=CWPKI2033E: The ACME service could not find the certificate for the {0} alias in the {1} keystore. The error is ''{2}''.
+#CWPKI2033E.explanation=A certificate authority signed certificate is stored in the default keystore and replaces any existing default certificate, but keystore or default certificate could not be accessed.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:158]: Change,
"A certificate authority signed certificate is stored in the default keystore and replaces any existing default certificate, but keystore or default certificate could not be accessed."
to,
"A certificate-authority-signed certificate is stored in the default keystore and replaces any existing default certificate. However, the keystore or default certificate cannot be accessed."

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops [hunk 2]
```diff
@@ -154,9 +154,9 @@ CWPKI2032E=CWPKI2032E: The certificate subject alternative names in the default
...
+#CWPKI2033E.useraction=Review the error message for details on the failure.
```
Comment (ManasiGandhi) [inline dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:159]: Change,
"Review the error message for details on the failure"
to,
"Review the error message for details"

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops [hunk 3]
```diff
@@ -154,9 +154,9 @@ CWPKI2032E=CWPKI2032E: The certificate subject alternative names in the default
 CWPKI2032E.explanation=The certificate is an invalid DER-encoded certificate or contains unsupported DER features.
 CWPKI2032E.useraction=Review the error message for details on the failure. Revoke or remove the invalid certificate.
 
-CWPKI2033E=CWPKI2033E: The ACME service could not find the certificate for the {0} alias in the {1} keystore. The error is ''{2}''.
-CWPKI2033E.explanation=A certificate authority signed certificate is stored in the default keystore and replaces any existing default certificate, but keystore or default certificate could not be accessed.
-CWPKI2033E.useraction=Review the error message for details on the failure.
+#CWPKI2033E=CWPKI2033E: The ACME service could not find the certificate for the {0} alias in the {1} keystore. The error is ''{2}''.
```
[the remark this answers is not in the document]
Comment (jvanhill) [reply dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:157]: This message has been commented out and is unused. It has only been left to make it clear that we have an open message.

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops (hunk 1 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:158]: This message has been commented out and is unused. It has only been left to make it clear that we have an open message.

File: dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.acme/resources/com/ibm/ws/security/acme/resources/AcmeMessages.nlsprops:159]: This message has been commented out and is unused. It has only been left to make it clear that we have an open message.

**Comment by jvanhill** [comment]
#libby

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
#libby

**Comment by jvanhill** [comment]
#libby
#build

**Comment by jvanhill** [comment]
#build

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

**Comment by jvanhill** [comment]
#libby
#build

**Comment by jvanhill** [comment]
Rebuilding since there were merge conflicts.

**Comment by jvanhill** [comment]
Build failed during upload. Re-spinning.

#build

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
#libby
#build

**Comment by jvanhill** [comment]
The lone failing test (for which there was a timeout and a javacore failure) was one for which there was an existing defect.

**Pull request merged**